### ISSUE TITLE
capture "non whitelisted IP" error

### ISSUE THREAD

**Issue by adamdecaf** [namecheap/go-namecheap-sdk#4]
I didn't see an error reported with, but got the following with `DEBUG=1` 

```xml
<?xml version="1.0" encoding="utf-8"?>
<ApiResponse Status="ERROR" xmlns="http://api.namecheap.com/xml.response">
  <Errors>
    <Error Number="1011150">Invalid request IP: 50.242.74.11</Error>
  </Errors>
  <Warnings />
  <RequestedCommand />
  <Server>PHX01APIEXT03</Server>
  <GMTTimeDifference>--5:00</GMTTimeDifference>
  <ExecutionTime>0</ExecutionTime>
</ApiResponse>
```

**Comment by caiconkhicon** [comment]
When running `terraform apply`, I received the error:
`Error: Failed to create namecheap Record: Could not find the record with hash 1893805056`

In my case, it was my public IP not being in the whitelist of my Namecheap account's API, but I assume that something like wrong token key would also return the same error. The error is quite misleading, because it looks like the record doesn't exist. IMO, there should be a dedicated error message for not being able to connect to the Namecheap API endpoint.

**Comment by ifree92** [comment]
The error: 

```xml
    <Error Number="1011150">Invalid request IP: 50.242.74.11</Error>
```

is possible only if you have not whitelisted your IP. Some details are [here](https://www.namecheap.com/support/api/intro/)

@caiconkhicon this error most likely resolved with #24
It depends on which exactly request you have been executing. Upon refactoring, we got rid of "hashing" the records so probably you won't face this error anymore :)
After merging the complete refactoring I will close this issue. You're welcome to reopen if that will still be relevant for you.
